pkg/utils/klog/klog.go

```diff
@@ -0,0 +1,72 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package klog
+
+import (
+	"flag"
+	"fmt"
+	"strings"
+
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/node-feature-discovery/pkg/utils"
+)
+
+// InitKlogFlags function is responsible for initializing klog flags.
+func InitKlogFlags(flagset *flag.FlagSet) map[string]*utils.KlogFlagVal {
+	klogFlags := make(map[string]*utils.KlogFlagVal)
+
+	flags := flag.NewFlagSet("klog flags", flag.ContinueOnError)
+	klog.InitFlags(flags)
+	flags.VisitAll(func(f *flag.Flag) {
+		name := klogConfigOptName(f.Name)
+		klogFlags[name] = utils.NewKlogFlagVal(f)
+		flagset.Var(klogFlags[name], f.Name, f.Usage)
+	})
+
+	return klogFlags
+}
+
+// HandleKlogConfiguration updates the klog configuration taking into consideration
+// klog cli options, and klog configuration specified in the config file.
+func HandleKlogConfiguration(klogArgs map[string]*utils.KlogFlagVal, klogConfig map[string]string) error {
```

Comment:
Just thinking aloud here: this function merges klog cmdline flags to klog configuration (file) options, basically overriding the config options with anything that was explicitly specified on cmdline. So, should we name this function `UpdateKlogConfigOpts` or `MergeKlogConfiguration` or smth 🤨

Another question: should we specify a separate type for the klog config opts, like
```golang
type KlogConfigOpts map[string]string
```

WDYT?

Reply:
@marquiz Sounds good to me! Updated!